app/src/org/commcare/dalvik/odk/provider/InstanceProvider.java

```diff
@@ -517,6 +537,10 @@ private void linkToSessionFormRecord(Uri instanceUri) {
             if (FormRecord.STATUS_COMPLETE.equals(current.getStatus())) {
                 try {
                     new FormRecordProcessor(getContext()).process(current);
+                } catch (InvalidStructureException | InvalidStorageStructureException e) {
+                    // record should be wiped when form entry is exited
+                    Logger.log(AndroidLogger.TYPE_ERROR_WORKFLOW, e.getMessage());
```

Comment:
I'm just copying this line from the catch directly below, which was the old code path